jsapp/js/utils.ts

```diff
@@ -532,3 +532,59 @@ export function getExponentialDelayTime(
 
   return count;
 }
+
+// Submission UUID comparison helpers
+
+/**
+ * Add a default 'uuid:' prefix to the provided identifier,
+ * if there's no prefix already.
+ *
+ *  Examples:
+ *    'kobotoolbox.org:123456789' -> unchanged
+ *    'uuid:123456789' -> unchanged
+ *    '123456789' -> 'uuid:123456789'
+ *
+ * 🐍 Equivalent to add_uuid_prefix on the backend.
+ */
+export function addDefaultUuidPrefix(uuid: string) {
+  return uuid.includes(':') ? uuid : `uuid:${uuid}`;
+}
+
+/**
+ * Remove the default 'uuid:' prefix from a provided identifier,
+ * while preserving any custom prefixes (e.g. 'kobotoolbox.org:123456789'),
+ * which are allowed by OpenRosa to reduce the chance of ID collisions.
+ *
+ * 🐍 Equivalent to remove_uuid_prefix on the backend.
+ */
+export function removeDefaultUuidPrefix(uuid: string) {
+  return uuid.replace(/^uuid:/, '');
+}
```

Comment:
xref [🐍 kobo/apps/openrosa/apps/logger/xform_instance_parser.py](https://github.com/kobotoolbox/kpi/blob/efd05b375cfa9565261d2f238d3d6ebd0af47b50/kobo/apps/openrosa/apps/logger/xform_instance_parser.py#L137-L145)

```py
def remove_uuid_prefix(uuid_: str) -> str:
    """
    Strips the 'uuid:' prefix from the provided identifier if it exists.
    This preserves any custom ID schemes (e.g., 'kobotoolbox.org:123456789')
    while ensuring only the 'uuid:' prefix is removed. This approach
    adheres to the OpenRosa spec, allowing custom prefixes to be stored
    intact in the database to prevent potential ID collisions.
    """
    return re.sub(r'^uuid:', '', uuid_)
```